### PR TITLE
o/snapstate: support ignore-validation flag when updating to a specific snap revision

### PR DESCRIPTION
**Pull request by stolowski**
Support --ignore-validation flag when requesting refresh to a specific revision and it doesn't match the revision required by validation set.

Followup to #10760

### CODE REVIEW

**Review by pedronis — APPROVED** [review]
thank you

File: overlord/snapstate/snapstate_update_test.go [hunk 1]
```diff
@@ -6528,6 +6528,79 @@ func (s *validationSetsSuite) TestUpdateToRevisionSnapRequiredByValidationWronRe
...
+		Sequence: []*snap.SideInfo{si},
+		Current:  snap.R(1),
+		SnapType: "app",
+		Flags: snapstate.Flags{
+			IgnoreValidation: true,
+		},
+	})
+	snaptest.MockSnap(c, `name: some-snap`, si)
+
+	// revision 5 is required ane requesting revision 11 would fail
```
Comment (MiguelPires) [inline overlord/snapstate/snapstate_update_test.go:6569]:
```suggestion
	// revision 5 is required and requesting revision 11 would fail
```

**Review by MiguelPires — APPROVED** [review]
LGTM

File: overlord/snapstate/snapstate_update_test.go [hunk 2]
```diff
@@ -6528,6 +6528,79 @@ func (s *validationSetsSuite) TestUpdateToRevisionSnapRequiredByValidationWronRe
 	c.Assert(err, ErrorMatches, `cannot update snap "some-snap" to revision 11 without --ignore-validation, revision 5 is required by validation sets: 16/foo/bar/2`)
 }
 
+func (s *validationSetsSuite) TestUpdateToRevisionSnapRequiredByValidationWrongRevisionIgnoreValidation(c *C) {
```
Comment (mardy) [inline overlord/snapstate/snapstate_update_test.go:6531]: nitpick: I'm a fan of long names myself :-), but there is a limit above which they become hard to understand, and I'm afraid this is one of them. In a case like this one I'd prefer to have a 2-3 lines comment that explains what this function does (if it's needed), or maybe several inline comments explaining the not obvious steps.

File: overlord/snapstate/snapstate_update_test.go [hunk 3]
```diff
@@ -6528,6 +6528,79 @@ func (s *validationSetsSuite) TestUpdateToRevisionSnapRequiredByValidationWronRe
...
+		Current:  snap.R(1),
+		SnapType: "app",
+		Flags: snapstate.Flags{
+			IgnoreValidation: true,
+		},
+	})
+	snaptest.MockSnap(c, `name: some-snap`, si)
+
+	// revision 5 is required and requesting revision 11 would fail
+	// without --ignore-validation.
```
Comment (mardy) [inline overlord/snapstate/snapstate_update_test.go:6570]: Thanks for this comment! My remaining doubt is, though, that the currently installed revision od `some-snap` is 1, which also does not satisfy the assertion; what would happen if the currently installed revision was 5? Would the `IgnoreValidation` flag help in that case, too?

(if so, I would probably change the test to set the currently installed revision to 5, in order to make that obvious)

**Review by mardy — APPROVED** [review]
LGTM, some question on the tests.

File: overlord/snapstate/snapstate_update_test.go (hunk 3 above)
Comment (stolowski) [reply overlord/snapstate/snapstate_update_test.go:6576]: This doesn't matter for the test, we could have this scenario after installing the snap with --ignore-validation before - I've added a comment explaining this.

File: overlord/snapstate/snapstate_update_test.go (hunk 2 above)
Comment (stolowski) [reply overlord/snapstate/snapstate_update_test.go]: Fair point, renamed with a comment. Thanks

**Pull request merged**